tcl/file.tcl

```diff
@@ -92,9 +92,10 @@ proc ::file::New {} {
 #    Opens file-open dialog and opens the selected Scid database.
 #
 proc ::file::Open {{fName ""}} {
-  set err [::file::Open_ "$fName"]
+  lassign [::file::Open_ "$fName"] err fName
```

Comment:
I believe it is better to move the `if {$fName == ""} {` into ::file::Open instead of returning a different fName.